libs/azure-iot-middleware-freertos/source/include/azure_iot_adu_client.h

```diff
@@ -135,10 +144,17 @@ typedef struct AzureIoTADUClientDeviceProperties
     const uint8_t * ucModel;
     uint32_t ulModelLength;
 
+    AzureIoTADUDeviceCustomProperties_t * pxCustomProperties;
+
     AzureIoTADUClientUpdateId_t xCurrentUpdateId;
 
     const uint8_t * ucDeliveryOptimizationAgentVersion;
     uint32_t ulDeliveryOptimizationAgentVersionLength;
+
+    struct
+    {
+        az_iot_adu_device_custom_properties xCustomProperties;
+    } _internal;
```

Comment:
should this go in here or in the `AzureIoTADUDeviceCustomProperties_t` struct? If it's there, the user only takes a hit of a pointer if there are no custom properties.

Reply:
God, obviously. Try doing this with a crying baby... :)

Reply:
Fixed.